src/main/java/io/reactiverse/pgclient/impl/PgConnectionUriParser.java

```diff
@@ -0,0 +1,221 @@
+package io.reactiverse.pgclient.impl;
+
+import io.reactiverse.pgclient.PgConnectOptions;
+import io.reactiverse.pgclient.PgPoolOptions;
+import io.vertx.core.json.JsonObject;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.Integer.*;
+import static java.lang.String.*;
+
+/**
+ * This is Parser for parsing connection URIs of PostgreSQL.
+ * Based on Version 9.6
+ *
+ * @author Billy Yuan <billy112487983@gmail.com>
+ */
+public class PgConnectionUriParser {
+  private static final String FULL_URI_REGEX = "^postgre(?:s|sql)://(?:(\\w+(?::\\S+)?)@)?([0-9.]+|\\[[A-Za-z0-9:]+]|[A-Za-z0-9.%\\-_]+)?(?::(\\d+))?(?:/([A-Za-z0-9_\\-]+))?(?:\\?(.*))?$";
+
+  private static int USER_INFO_GROUP = 1;
+  private static int NET_LOCATION_GROUP = 2;
+  private static int PORT_GROUP = 3;
+  private static int DATABASE_GROUP = 4;
+  private static int PARAMETER_GROUP = 5;
+
+  private final String connectionUri;
+
+  private JsonObject configuration = new JsonObject();
+
+  private PgConnectionUriParser(String connectionUri) {
```

Comment:
make everything static  and pass the object along the way in methods

Reply:
changes are in https://github.com/reactiverse/reactive-pg-client/pull/93/commits/92afd499d8a45971da47ebfb5ddf23fc505c176f.